### ISSUE TITLE
create an aim plugin

### ISSUE THREAD

**Issue by kolodny** [gabrielbull/react-aim#1]
I don't really see why this needs to use react (unless I'm missing something). Would it make sense to abstract this away to an `aim` plugin and have a minimal `react-aim` wrapper?

**Comment by gabrielbull** [comment]
You're right that the logic part of this library is not react specific, only the implementation. It would indeed be possible to extract the logic and create a separate library for it. I'll leave this issue open as a reminder to do this in the future, as it would require a bit of work.
